examples/dummy_honest/examples/dummy_honest.rs

```diff
@@ -0,0 +1,217 @@
+use aleph_bft::{run_session, Recipient};
+use aleph_bft_mock::{
+    Data, DataProvider, FinalizationHandler, Hasher64, Keychain, Loader, PartialMultisignature,
+    Saver, Signature, Spawner,
+};
+use chrono::Local;
+use clap::Parser;
+use codec::{Decode, Encode};
+use futures::{
+    channel::{mpsc, oneshot},
+    FutureExt, StreamExt,
+};
+use libp2p::{
+    development_transport,
+    floodsub::{self, Floodsub, FloodsubEvent},
+    identity,
+    mdns::{Mdns, MdnsConfig, MdnsEvent},
+    swarm::{NetworkBehaviourEventProcess, SwarmBuilder},
+    NetworkBehaviour, PeerId, Swarm,
+};
+use log::{debug, info, warn};
+use parking_lot::Mutex;
+use std::{error::Error, io::Write, sync::Arc, time::Duration};
+
+const ALEPH_PROTOCOL_NAME: &str = "aleph";
+
+/// Dummy honest node example.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// Our index
+    #[clap(long)]
+    my_id: usize,
+
+    /// Size of the committee
+    #[clap(long)]
+    n_members: usize,
+
+    /// Number of data to be finalized
+    #[clap(long)]
+    n_finalized: usize,
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::builder()
+        .format(|buf, record| {
+            writeln!(
+                buf,
+                "{} {}: {}",
+                record.level(),
+                Local::now().format("%Y-%m-%d %H:%M:%S%.3f"),
+                record.args()
+            )
+        })
+        .filter(None, log::LevelFilter::Debug)
+        .init();
+
+    let args = Args::parse();
+
+    info!(target: "dummy-honest", "Getting network up.");
+    let (network, mut manager) = Network::new().await.unwrap();
+    let (close_network, exit) = oneshot::channel();
+    tokio::spawn(async move { manager.run(exit).await });
+
+    let data_provider = DataProvider::new();
+    let (finalization_handler, mut finalized_rx) = FinalizationHandler::new();
+
+    let units = Arc::new(Mutex::new(vec![]));
+    let unit_loader = Loader::new((*units.lock()).clone());
```

Comment:
Yes, by doing it this way, the vector is not quite shared between loader and saver. If we wanted to make it right I think one would need to fix the mocks, so that you can do something like 
`let loader, saver = new_loader_and_saver()` and `new_loader_and_saver` would create these sharing a Mutex wrapped `Vec`.

Reply:
But saver and loader should not share a mutex. Saver should have no impact on data stored in loader and loader should have data backed up before crash, so in this case empty vector.

Reply:
Yeah, you are right, you can disregard what I said above :)